From 94dd798fb4c01e52eda4e166e2a839e4b338e524 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Thu, 5 Sep 2019 17:44:15 -0700
Subject: [PATCH 0986/1921] drivers: cleanup el3 handler only if parent process
 exits

We have EL0 interrupt handling in various drivers. There are cleanup
handlers that are called when a process is killed. These handlers are
also being called when child process exits. So do not cleanup if its
child.

Change-Id: Ie068860174c71155283482d414184abcd04728b6
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15390
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/18209
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit 3c1276ffc89556880183f9fa5fb30491214adb91)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42619
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/gpio/gpio-thunderx.c | 3 +--
 drivers/misc/otx_bphy_ctr.c  | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index c65d25068fbd..08d39be0a6d1 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -232,8 +232,7 @@ static void cleanup_el3_irqs(struct task_struct *task)
 	for (i = 0; i < MAX_GPIO; i++) {
 		if (gpio_installed[i] &&
 		    gpio_installed_tasks[i] &&
-		    ((gpio_installed_tasks[i] == task) ||
-			(gpio_installed_tasks[i] == task->group_leader))) {
+		    (gpio_installed_tasks[i] == task)) {
 			pr_alert("Exiting, removing handler for GPIO %d\n",
 				 i);
 			__remove_el3_inthandler(i);
diff --git a/drivers/misc/otx_bphy_ctr.c b/drivers/misc/otx_bphy_ctr.c
index 1135d15850f4..07591cbb28df 100644
--- a/drivers/misc/otx_bphy_ctr.c
+++ b/drivers/misc/otx_bphy_ctr.c
@@ -168,8 +168,7 @@ static void cleanup_el3_irqs(struct task_struct *task)
 	for (i = 0; i < MAX_IRQ; i++) {
 		if (irq_installed[i] &&
 		    irq_installed_tasks[i] &&
-		    ((irq_installed_tasks[i] == task) ||
-			(irq_installed_tasks[i] == task->group_leader))) {
+		    (irq_installed_tasks[i] == task)) {
 			pr_alert("Exiting, removing handler for BPHY IRQ %d\n",
 				 i);
 			__remove_el3_inthandler(i);
-- 
2.31.1

